### PR TITLE
feat(webpack-cli): add no-hot flag

### PR DESCRIPTION
**Pull request by snitin315**
**What kind of change does this PR introduce?**

Added `--no-hot` flag to CLI.

**Did you add tests for your changes?**

YES

**If relevant, did you update the documentation?**

YES

**Summary**

Refers #1240 

Refers #1012 

**Does this PR introduce a breaking change?**

NO

**Other information**

### CODE REVIEW

File: packages/webpack-cli/lib/utils/cli-flags.js [hunk 1]
```diff
@@ -230,6 +230,14 @@ module.exports = {
             description: 'Enables Hot Module Replacement',
             link: 'https://webpack.js.org/concepts/hot-module-replacement/',
         },
+        {
+            name: 'no-hot',
+            usage: '--no-hot',
```
Comment (ematipico) [inline packages/webpack-cli/lib/utils/cli-flags.js]: `webpack --no-hot`

File: packages/webpack-cli/lib/groups/AdvancedGroup.js [hunk 2]
```diff
@@ -57,6 +57,12 @@ class AdvancedGroup extends GroupHelper {
                 options.plugins = [hotModuleVal];
```
Comment (ematipico) [inline packages/webpack-cli/lib/groups/AdvancedGroup.js]: Could you also create a unit test inside `AdvancedGroup.test.js` file? Similar to `ZeroConfigGroup.test.js`

File: test/no-hot/no-hot.test.js [hunk 3]
```diff
@@ -0,0 +1,17 @@
+'use strict';
+const { run } = require('../utils/test-utils');
+const { stat } = require('fs');
+const { resolve } = require('path');
+describe('no-hot flag', () => {
```
Comment (ematipico) [inline test/no-hot/no-hot.test.js]: Could you also create a test where we have a config file with `hot: true` and you provide `--no-hot`?

Comment (ematipico) [reply test/no-hot/no-hot.test.js]: Also, a test with just `--no-hot`. It should be successful as test.

File: test/no-hot/no-hot.test.js [hunk 4]
```diff
@@ -0,0 +1,28 @@
...
+    it('should be successful when --no-hot is passed', done => {
+        const { stderr, stdout } = run(__dirname, ['--no-hot']);
+        expect(stderr).toHaveLength(0);
```
Comment (ematipico) [inline test/no-hot/no-hot.test.js]: `toBeFalsy`

**Comment by snitin315** [comment]
> a test for the serve command (webpack-dev-server consumes the flag)

@ematipico how to write a test for this one, it will be great if you can provide me an overview. Thanks!

File: packages/webpack-cli/__tests__/AdvancedGroup.test.js [hunk 5]
```diff
@@ -0,0 +1,14 @@
+const AdvancedGroup = require('../lib/groups/AdvancedGroup');
+
+describe('GroupHelper', function() {
+    it('should  load the hot advanced config', () => {
```
Comment (jamesgeorge007) [inline packages/webpack-cli/__tests__/AdvancedGroup.test.js]:
```suggestion
    it('should load the hot advanced config', () => {
```

Comment (jamesgeorge007) [reply packages/webpack-cli/__tests__/AdvancedGroup.test.js]: An extra space :sweat_smile:

Comment (snitin315) [reply packages/webpack-cli/__tests__/AdvancedGroup.test.js]: oops!

**Pull request closed without merging**